azurerm/internal/services/securitycenter/azurerm_security_center_auto_provisioning.go

```diff
@@ -0,0 +1,113 @@
+		Delete: resourceArmSecurityCenterAutoProvisioningDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"auto_provision": {
+				Type:     schema.TypeString,
+				Required: true,
+				// NOTE: the API seems case insensitive to this string value, 'ON', 'On', 'on' all work
+				ValidateFunc: validation.StringInSlice([]string{
+					string(security.AutoProvisionOn),
+					string(security.AutoProvisionOff),
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceArmSecurityCenterAutoProvisioningUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.AutoProvisioningClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := securityCenterAutoProvisioningName
+
+	// No need for import check as there's always single resource called 'default'
+	// - it cannot be deleted, all this does is set a string property to: "on" or "off"
+
+	// Build settings struct with auto_provision value
+	settings := security.AutoProvisioningSetting{
+		AutoProvisioningSettingProperties: &security.AutoProvisioningSettingProperties{
+			AutoProvision: security.AutoProvision(d.Get("auto_provision").(string)),
+		},
+	}
+
+	// There is no update function or operation in the API, only create
+	if _, err := client.Create(ctx, name, settings); err != nil {
+		return fmt.Errorf("Error creating/updating Security Center auto provisioning: %+v", err)
+	}
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		return fmt.Errorf("Error reading Security Center auto provisioning: %+v", err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Security Center auto provisioning ID is nil")
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmSecurityCenterAutoProvisioningRead(d, meta)
+}
+
+func resourceArmSecurityCenterAutoProvisioningRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).SecurityCenter.AutoProvisioningClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resp, err := client.Get(ctx, securityCenterAutoProvisioningName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] Security Center subscription was not found: %v", err)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error reading Security Center auto provisioning: %+v", err)
+	}
+
+	if properties := resp.AutoProvisioningSettingProperties; properties != nil {
+		d.Set("auto_provision", properties.AutoProvision)
+	}
+
+	return nil
+}
+
+func resourceArmSecurityCenterAutoProvisioningDelete(_ *schema.ResourceData, _ interface{}) error {
```

Comment:
Is there a "default" state that we should return this to? e.g. if it defaults to `Off`, and the user has it previously configured `On`, would the expectation from the user be that it is switched `Off` by deletion of the resource?